Kubernetes/HOWTO-on-prem.md

```diff
@@ -102,91 +75,219 @@ sudo cp -r !(*.tgz) ${CNI_BIN}
+
+
+## Deploying On Existing Clusters ##
+This section applies to people who want to add nodes to an existing Linux cluster that they've painstakingly set up. Whether set up manually, or via `kubeadm`, the only thing that you need from the Linux master is the _certificate configuration file_.
+
+If you used `kubeadm`, this would probably be at `/etc/kubernetes/admin.conf`. If you deployed manually, this may be at `~/.kube/config`. Wherever it is, just drop a copy at `C:\k\config` on the Windows node and things should work.
+
+Of course, you will still need to [build the Windows binaries](#building-kubernetes-windows-binaries) on a Linux box.
+
+
+## Building Kubernetes' Windows Binaries ##
+We will need to build the `kubelet` and `kubeproxy` binaries for Windows from scratch by _cross-compiling from Linux_. There are multiple ways to do this:
+
+  - Build them [locally](#build-locally).
+  - Generate the binaries using [Vagrant](#build-with-vagrant).
+  - Leverage the [standard containerized build scripts](https://github.com/kubernetes/kubernetes/tree/master/build#key-scripts) in the Kubernetes project. For this, follow the steps for [building locally](#build-locally) up to the `make` steps, then use the linked instructions.
+
+**Note**: If you run into "permission denied" errors, these can be avoided by building the Linux `kubelet` first, per the note in [acs-engine](https://github.com/Azure/acs-engine/blob/master/scripts/build-windows-k8s.sh#L176):
+
+> Due to what appears to be a bug in the Kubernetes Windows build system, one has to first build a Linux binary to generate `_output/bin/deepcopy-gen`. Building to Windows w/o doing this will generate an empty `deepcopy-gen`.
+
+Additionally, you will need to download the `kubectl.exe` binary. The [release notes](https://github.com/kubernetes/kubernetes/releases/tag/v1.8.2) have links in the `CHANGELOG-1.8.md` file. Copy these to a new directory, `~/kube-win`. We'll need to transfer them to the Windows node later:
+
+```bash
+mkdir ~/kube-win
+wget -O kubernetes-windows.tar.gz https://dl.k8s.io/v1.8.2/kubernetes-client-windows-amd64.tar.gz
+tar -vxzf kubernetes-windows.tar.gz 
+cp kubernetes/client/bin/kubectl.exe ~/kube-win/
+```
+
+
+#### Build Locally ####
+Set up a [Go environment](https://golang.org/doc/install#tarball); don't forget to set your `$GOPATH`! Then, run these commands to build:
+
+```bash
+$ KUBEREPO="k8s.io/kubernetes"
+$ go get -d $KUBEREPO
+# Note: the above command may spit out a message about 
+#       "no Go files in...", but it can be safely ignored!
+$ cd $GOPATH/src/$KUBEREPO
+# if you want the "bleeding edge" Kubernetes, run 
+# `git checkout release-1.8` here.
+$ git checkout tags/v1.8.2
+$ make clean && make WHAT=cmd/kubelet
+
+# necessary to add a bug-fix not yet upstream'd
+$ git add remote patch https://github.com/madhanrm/kubernetes
+$ git fetch patch

```

Comment:
Is this patch specific to v1.8.2 or will it work with latest upstream (and latest alpha build?)

Reply:
Once the PR you linked goes in, it won't be necessary. Otherwise, it's always needed.